ui/src/shared/components/Dygraph.js

```diff
@@ -65,18 +71,26 @@ export default class Dygraph extends Component {
 
     const graphRef = this.graphRef
     const legendRef = this.legendRef
-    let finalLineColors = overrideLineColors
+    const finalLineColors = [...(overrideLineColors || LINE_COLORS)]
+
+    const hashColorDygraphSeries = {}
+    const {length} = finalLineColors
```

Comment:
If this truly is a constant set, a better approach might be to use a perfect hashing function and regenerate it if the set changes in the future. That would also resolve the distribution issues mentioned earlier.